aptos-move/e2e-move-tests/src/tests/resource_account.rs

```diff
@@ -0,0 +1,193 @@
+use crate::{assert_success, tests::common, MoveHarness};
+use aptos_types::{
+    account_address::{create_resource_address, AccountAddress},
+    event::EventHandle,
+    transaction::{EntryFunction, TransactionPayload},
+};
+use cached_packages::aptos_stdlib;
+use framework::{BuildOptions, BuiltPackage};
+use language_e2e_tests::account::Account;
+use move_deps::move_core_types::{
+    ident_str, identifier::Identifier, language_storage::ModuleId, parser::parse_struct_tag,
+};
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize)]
+struct ModuleData {
+    resource_signer_cap: AccountAddress,
+}
+
+#[derive(Serialize, Deserialize)]
+struct Coin {
+    value: u64,
+}
+
+#[derive(Serialize, Deserialize)]
+struct CoinStore {
+    coin: Coin,
+    frozen: bool,
+    deposit_events: EventHandle,
+    withdraw_events: EventHandle,
+}
```

Comment:
Please use the coinstore in `types/src/account_config/resources/coin_store.rs` and maybe update it to be more general?